### PR TITLE
Xgboost-quickstart example

### PR DESCRIPTION
**Pull request by yan-gao-GY**
### Description

We need a simpler quick-start example for XGBoost, which allows users to get a quick idea of how to use XGBoost in Flower. 

## Proposal

In this example, we simply the data partitioning part, and remove the customised experimental settings. We only use two clients for the training. The evaluation is conducted on clients' hold-out test set.

### Note

After bagging strategy has been moved to the core framework, this [PR](https://github.com/adap/flower/pull/2611), we need to change the server code here to use the strategy from the core framework.

### CODE REVIEW

File: examples/xgboost-quickstart/README.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
+# Flower Example using XGBoost
+
+This example demonstrates how to perform EXtreme Gradient Boosting (XGBoost) within Flower using `xgboost` package.
+Tree-based with bagging method is used for aggregation on the server.
+
```
Comment (danieljanes) [inline examples/xgboost-quickstart/README.md:5]:
```suggestion
Tree-based with bagging method is used for aggregation on the server.

This project provides a minimal code example to enable you to get stated quickly. For a more comprehensive code example, take a look at [xgboost-comprehensive](https://github.com/adap/flower/tree/main/examples/xgboost-comprehensive).
```

File: examples/xgboost-quickstart/README.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+## Project Setup
+
+Start by cloning the example project. We prepared a single-line command that you can copy into your shell which will checkout the example for you:
+
+```shell
+git clone --depth=1 https://github.com/adap/flower.git && mv flower/examples/quickstart-xgboost . && rm -rf flower && cd quickstart-xgboost
+```
+
+This will create a new directory called `quickstart-xgboost` containing the following files:
```
Comment (danieljanes) [inline examples/xgboost-quickstart/README.md:14]: There are a few more instances of this below:
```suggestion
This will create a new directory called `xgboost-quickstart` containing the following files:
```

**Review by danieljanes — CHANGES_REQUESTED** [review]
Example looks great. To further simplify, I'd suggest to:
- Remove `strategy.py` and use the strategy in `flwr.server.strategy`
- Remove `dataset.py` and move the code into `client.py`

File: examples/xgboost-quickstart/client.py [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+    new_data = xgb.DMatrix(x, label=y)
+    return new_data
+
+
+# Load (HIGGS) dataset and conduct partitioning
+partitioner = IidPartitioner(num_partitions=10)
+fds = FederatedDataset(dataset="jxie/higgs", partitioners={"train": partitioner})
+
+# Let's use the first partition as an example
+partition = fds.load_partition(idx=0, split="train")
```
Comment (danieljanes) [inline examples/xgboost-quickstart/client.py:53]: Should we add the possibility to load a specific partition by providing `--node-id 0` or `--node-id 1`? And then use `IidPartitioner(num_partitions=2)` above?

Comment (yan-gao-GY) [reply examples/xgboost-quickstart/client.py:53]: ok, i'll pull back the arguments parser only for `--node-id`

File: examples/xgboost-quickstart/client.py [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+    y = data["label"]
+    new_data = xgb.DMatrix(x, label=y)
+    return new_data
+
+
+# Load (HIGGS) dataset and conduct partitioning
+partitioner = IidPartitioner(num_partitions=2)
+fds = FederatedDataset(dataset="jxie/higgs", partitioners={"train": partitioner})
+
+# Let's use the first partition as an example
```
Comment (danieljanes) [inline examples/xgboost-quickstart/client.py:63]:
```suggestion
# Load the partition for this `node_id`
```

**Review by danieljanes — APPROVED** [review]
